### PR TITLE
fix(popup): Put Popup in Parent hierarchy of its Child content

### PR DESCRIPTION
**Pull request by davidjohnoliver**
GitHub Issue (If applicable): fixes https://github.com/unoplatform/uno/issues/5852

Set Popup as Parent of PopupPanel. This emulates UWP behavior that the Popup is a Parent of one of the parents of the Child content, which is relied upon in some cases (eg WCT DataGrid relies upon this for focus management).

## PR Type

What kind of change does this PR introduce?

- Bugfix

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs [hunk 1]
```diff
@@ -174,17 +174,18 @@ protected override Size ArrangeOverride(Size finalSize)
 			return finalSize;
 		}
 
-#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:177]: Why was this added to be android only, originally ?

Comment (davidjohnoliver) [reply src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs:177]: I actually added it a couple of days ago to fix ComboBox theme updates, I limited it to Android out of conservatism because I only actually needed it on Android (the Parent was being set to the native host view but not being unset).

**Comment by jeromelaban** [comment]
Should we add runtime test for this one ?

**Review by jeromelaban — APPROVED** [review]
